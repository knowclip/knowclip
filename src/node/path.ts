export { basename, extname, dirname, resolve, join } from 'path'
