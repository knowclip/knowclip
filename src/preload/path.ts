import { getPreloadModule } from './getPreloadModule'

const { basename, extname, dirname, resolve, join } =
  getPreloadModule('path') || require('path')

export { basename, extname, dirname, resolve, join }
