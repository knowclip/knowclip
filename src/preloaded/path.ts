import { getPreloadModule } from './getPreloadModule'

const { basename, extname, dirname, resolve, join } = getPreloadModule('path')

export { basename, extname, dirname, resolve, join }
