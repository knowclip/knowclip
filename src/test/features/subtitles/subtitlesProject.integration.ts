import {
  startApp,
  stopApp,
  ASSETS_DIRECTORY,
  GENERATED_ASSETS_DIRECTORY,
  initTestContext,
} from '../../setUpDriver'
import makeCardsFromSubtitles from './makeCardsFromSubtitles'
import saveAndCloseProject from './saveAndCloseProject'
import { join } from 'path'
import { mockSideEffects } from '../../mockSideEffects'
import { runAll, step } from '../step'
import { describe, beforeAll, afterAll } from 'vitest'

const testId = 'subtitlesProject'

describe('make clips and cards from subtitles', () => {
  const context = initTestContext(testId)

  beforeAll(async () => {
    const { app } = await startApp(
      context,
      persistedState(context.temporaryDirectory)
    )

    await mockSideEffects(app, sideEffectsMocks)
  })

  runAll(
    [
      step('automatically make clips and cards from subtitles', (context) =>
        makeCardsFromSubtitles(context)
      ),
      step('saves and closes project', (context) =>
        saveAndCloseProject(context)
      ),
    ],
    context
  )

  afterAll(async () => {
    await stopApp(context)
  })
})

const sideEffectsMocks = {
  uuid: [
    '94367f0c-92ef-4542-a80a-6fbaf8e1ad76',
    'e78620d8-9ebd-45b8-b335-d3280a4f3ecc',
    '320265ad-8190-4729-8ec1-5fcf0d1b832e',
    '2f5ad340-c06b-4754-a468-56852ec9638c',
    '7d4e2afe-55d1-4a1a-8365-767aa83c83e5',
    '8a0d55ba-e456-4108-9fe2-911d7d25ffde',
    '193c54f4-49f0-442b-b866-320ba111d369',
    'c7e5886a-0899-4192-8a9c-4b30c304a717',
    'f4f17e82-af87-459e-af11-4f3d65f9d5d7',
    'f1def124-bbaa-418a-8836-34a8d8cf74da',
    '6c34f168-1e1d-4933-84ae-df589c68ec06',
    '24374cd7-bed9-4f30-aace-dbdf8fe31422',
    '8f0f15ae-fe46-4b73-ad7d-36c4d10359dd',
    '2a3ce4c2-4714-47f6-9e33-2e28ba56eb1f',
    '90bf45fc-00e0-41b7-b82e-fda29473761e',
    'f85a53cc-4763-41a7-bfb2-ef615e2fc84d',
    '82746bca-e8a1-4411-9b0c-39a0085526ca',
    '78ec78a2-8520-4311-9f15-9f2061e6ffe5',
    '8a27cad2-517a-47fb-a68d-4e527bec6387',
  ],
  nowUtcTimestamp: [
    '2020-01-24T23:44:33Z',
    '2020-01-24T23:44:38Z',
    '2020-02-02T15:29:14Z',
    '2020-02-02T15:29:14Z',
    '2020-02-02T15:29:15Z',
    '2020-02-02T15:29:15Z',
    '2020-02-02T15:29:15Z',
    '2020-02-02T15:29:17Z',
    '2020-02-02T15:29:17Z',
    '2020-02-02T15:29:19Z',
    '2020-02-02T15:29:19Z',
    '2020-02-02T15:29:19Z',
    '2020-02-02T15:29:19Z',
    '2020-02-02T15:29:19Z',
    '2020-02-02T15:29:19Z',
  ],
}

const persistedState = (tmpDirectory: string): Partial<AppState> => ({
  fileAvailabilities: {
    ProjectFile: {
      'ef3a2602-37a0-4158-89d6-47b75edb5bea': {
        type: 'ProjectFile',
        id: 'ef3a2602-37a0-4158-89d6-47b75edb5bea',
        parentId: null,
        name: 'Project with subtitles',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(tmpDirectory, '/project_with_subtitles.kyml'),
        lastOpened: '2020-02-01T16:11:16Z',
      },
    },
    MediaFile: {
      'bd8aca85-de21-4666-bbab-a33fd21d03f1': {
        type: 'MediaFile',
        id: 'bd8aca85-de21-4666-bbab-a33fd21d03f1',
        parentId: 'ef3a2602-37a0-4158-89d6-47b75edb5bea',
        name: 'polar_bear_cafe.mp4',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(ASSETS_DIRECTORY, '/polar_bear_cafe.mp4'),
        lastOpened: '2020-02-01T16:11:18Z',
      },
      'f5053503-8e5f-45c3-9afc-26f5e6eab8a8': {
        type: 'MediaFile',
        id: 'f5053503-8e5f-45c3-9afc-26f5e6eab8a8',
        parentId: 'ef3a2602-37a0-4158-89d6-47b75edb5bea',
        name: 'piggeldy_cat.mp4',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(ASSETS_DIRECTORY, '/piggeldy_cat.mp4'),
        lastOpened: '2020-02-01T16:11:17Z',
      },
    },
    ExternalSubtitlesFile: {
      'a8b64692-dba0-4ef4-b5e7-8c887956a69a': {
        type: 'ExternalSubtitlesFile',
        id: 'a8b64692-dba0-4ef4-b5e7-8c887956a69a',
        parentId: 'bd8aca85-de21-4666-bbab-a33fd21d03f1',
        name: 'pbc_jp.ass',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(ASSETS_DIRECTORY, '/pbc_jp.ass'),
        lastOpened: '2020-02-01T16:11:18Z',
      },
    },
    VttConvertedSubtitlesFile: {
      '5f581055-dbf5-4f3b-bcee-a3215e405ffc': {
        type: 'VttConvertedSubtitlesFile',
        id: '5f581055-dbf5-4f3b-bcee-a3215e405ffc',
        parentId: 'bd8aca85-de21-4666-bbab-a33fd21d03f1',
        name: 'VttConvertedSubtitlesFile',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe.mp4_2_5f581055-dbf5-4f3b-bcee-a3215e405ffc.vtt'
        ),
        lastOpened: '2020-02-01T16:11:27Z',
      },
      'a8b64692-dba0-4ef4-b5e7-8c887956a69a': {
        type: 'VttConvertedSubtitlesFile',
        id: 'a8b64692-dba0-4ef4-b5e7-8c887956a69a',
        parentId: 'a8b64692-dba0-4ef4-b5e7-8c887956a69a',
        name: 'VttConvertedSubtitlesFile',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/pbc_jp.ass_a8b64692-dba0-4ef4-b5e7-8c887956a69a.vtt'
        ),
        lastOpened: '2020-02-01T16:11:24Z',
      },
    },
    WaveformPng: {
      'bd8aca85-de21-4666-bbab-a33fd21d03f1': {
        type: 'WaveformPng',
        id: 'bd8aca85-de21-4666-bbab-a33fd21d03f1',
        parentId: 'bd8aca85-de21-4666-bbab-a33fd21d03f1',
        name: 'WaveformPng',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe.mp4_bd8aca85-de21-4666-bbab-a33fd21d03f1.png'
        ),
        lastOpened: '2020-02-01T16:11:19Z',
      },
      'f5053503-8e5f-45c3-9afc-26f5e6eab8a8': {
        type: 'WaveformPng',
        id: 'f5053503-8e5f-45c3-9afc-26f5e6eab8a8',
        parentId: 'f5053503-8e5f-45c3-9afc-26f5e6eab8a8',
        name: 'WaveformPng',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/piggeldy_cat.mp4_f5053503-8e5f-45c3-9afc-26f5e6eab8a8.png'
        ),
        lastOpened: '2020-02-01T16:11:17Z',
      },
    },
    VideoStillImage: {
      '6b02575a-0a4f-402a-8ea3-31f457cb5d8c': {
        type: 'VideoStillImage',
        id: '6b02575a-0a4f-402a-8ea3-31f457cb5d8c',
        parentId: null,
        name: 'VideoStillImage',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe_10-280_6b02575a-0a4f-402a-8ea3-31f457cb5d8c.png'
        ),
        lastOpened: '2020-02-01T16:10:42Z',
      },
      'ff54865f-ef52-4f16-9f20-2ae257eceb40': {
        type: 'VideoStillImage',
        id: 'ff54865f-ef52-4f16-9f20-2ae257eceb40',
        parentId: null,
        name: 'VideoStillImage',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe_16-760_ff54865f-ef52-4f16-9f20-2ae257eceb40.png'
        ),
        lastOpened: '2020-02-01T16:10:44Z',
      },
      'a3a0c3eb-3841-4d2f-851a-5af95cd14855': {
        type: 'VideoStillImage',
        id: 'a3a0c3eb-3841-4d2f-851a-5af95cd14855',
        parentId: null,
        name: 'VideoStillImage',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe_78-180_a3a0c3eb-3841-4d2f-851a-5af95cd14855.png'
        ),
        lastOpened: '2020-02-01T16:10:54Z',
      },
      '152077c6-336b-47d6-88e6-1ba4410db553': {
        type: 'VideoStillImage',
        id: '152077c6-336b-47d6-88e6-1ba4410db553',
        parentId: null,
        name: 'VideoStillImage',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/piggeldy_cat_7-900_152077c6-336b-47d6-88e6-1ba4410db553.png'
        ),
        lastOpened: '2020-02-01T16:11:17Z',
      },
      '864d822f-d482-438e-9620-684b2730a688': {
        type: 'VideoStillImage',
        id: '864d822f-d482-438e-9620-684b2730a688',
        parentId: null,
        name: 'VideoStillImage',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/piggeldy_cat_19-220_864d822f-d482-438e-9620-684b2730a688.png'
        ),
        lastOpened: '2020-02-01T16:11:17Z',
      },
      'c9723266-41d2-4676-884f-a811ec89c786': {
        type: 'VideoStillImage',
        id: 'c9723266-41d2-4676-884f-a811ec89c786',
        parentId: null,
        name: 'VideoStillImage',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/piggeldy_cat_43-020_c9723266-41d2-4676-884f-a811ec89c786.png'
        ),
        lastOpened: '2020-02-01T16:11:18Z',
      },
    },
    Dictionary: {},
  },
})
