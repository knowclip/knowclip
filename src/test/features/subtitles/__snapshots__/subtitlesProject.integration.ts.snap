// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`make clips and cards from subtitles > saves and closes project > project file was saved 1`] = `
{
  "value": {
    "media": [
      {
        "clips": [
          {
            "end": "00:12.47",
            "fields": {
              "meaning": "Chilling and eating bamboo 
grass is the best thing ever.",
              "transcription": "笹を食べながらのんびりするのは最高だなぁ",
            },
            "id": "320265ad-8190-4729-8ec1-5fcf0d1b832e",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:07.63",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:17.65",
            "fields": {
              "meaning": "Hey now. Look at you, being so lazy 
so early in the day.",
              "transcription": "ちょっと  またこの子は昼間からゴロゴロして",
            },
            "id": "2f5ad340-c06b-4754-a468-56852ec9638c",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:12.78",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:21.0",
            "fields": {
              "meaning": "You're in the way of my cleaning. Scoot.",
              "transcription": "掃除の邪魔よ  どきなさい",
            },
            "id": "7d4e2afe-55d1-4a1a-8365-767aa83c83e5",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:18.33",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:22.34",
            "fields": {
              "meaning": "Okay.",
              "transcription": "はいー",
            },
            "id": "8a0d55ba-e456-4108-9fe2-911d7d25ffde",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:21.0",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:27.0",
            "fields": {
              "meaning": "Seconds please, Mom.",
              "transcription": "お母さん  おかわり",
            },
            "id": "193c54f4-49f0-442b-b866-320ba111d369",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:24.8",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:32.19",
            "fields": {
              "meaning": "Exactly what do you have 
planned for your future?",
              "transcription": "どうするの  あなた将来は",
            },
            "id": "c7e5886a-0899-4192-8a9c-4b30c304a717",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:29.32",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:38.73",
            "fields": {
              "meaning": "Jeez... Could we not talk about my future when I'm relaxing?",
              "transcription": "もう  ゴロゴロしてるときに  将来の話なんかしないでよ",
            },
            "id": "f4f17e82-af87-459e-af11-4f3d65f9d5d7",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:33.23",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:42.46",
            "fields": {
              "meaning": "If you have so much time to relax, you could go look for a part-time job.",
              "transcription": "暇があるなら  ちょっとはバイトでも探しなさい",
            },
            "id": "f1def124-bbaa-418a-8836-34a8d8cf74da",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:38.87",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:49.18",
            "fields": {
              "meaning": "I'm too busy relaxing, 
so I'll do it later.",
              "transcription": "ええ  ゴロゴロするのに忙しいから  後にするよ",
            },
            "id": "6c34f168-1e1d-4933-84ae-df589c68ec06",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:42.73",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:55.1",
            "fields": {
              "meaning": "Don't try to suck me up!",
              "transcription": "ああー  吸わないで",
            },
            "id": "24374cd7-bed9-4f30-aace-dbdf8fe31422",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:50.33",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:57.67",
            "fields": {
              "meaning": "Jeez...",
              "transcription": "もう",
            },
            "id": "8f0f15ae-fe46-4b73-ad7d-36c4d10359dd",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:56.88",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "01:04.63",
            "fields": {
              "meaning": "I wonder why Mom can't understand how much time relaxing takes up.",
              "transcription": "どうしてお母さんはゴロゴロする忙しさが分かんないんだろな",
            },
            "id": "2a3ce4c2-4714-47f6-9e33-2e28ba56eb1f",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:59.02",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "01:21.8",
            "fields": {
              "meaning": "Excuse me, which of these will have jobs where I can get paid to do nothing?",
              "transcription": "えーと  働かないでお金がもらえる仕事が載っているのはどれですか",
            },
            "id": "82746bca-e8a1-4411-9b0c-39a0085526ca",
            "image": {
              "type": "VideoStill",
            },
            "start": "01:15.29",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "01:24.76",
            "fields": {
              "meaning": "I don't think there is one.",
              "transcription": "ないと思います",
            },
            "id": "78ec78a2-8520-4311-9f15-9f2061e6ffe5",
            "image": {
              "type": "VideoStill",
            },
            "start": "01:23.44",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "01:26.43",
            "fields": {
              "meaning": "Really?",
              "transcription": "え",
            },
            "id": "8a27cad2-517a-47fb-a68d-4e527bec6387",
            "image": {
              "type": "VideoStill",
            },
            "start": "01:25.25",
            "tags": [
              "pbc",
            ],
          },
        ],
        "duration": "01:27.062",
        "flashcardFieldsToSubtitlesTracks": {
          "meaning": "5f581055-dbf5-4f3b-bcee-a3215e405ffc",
          "transcription": "94367f0c-92ef-4542-a80a-6fbaf8e1ad76",
        },
        "format": "mov,mp4,m4a,3gp,3g2,mj2",
        "id": "bd8aca85-de21-4666-bbab-a33fd21d03f1",
        "name": "polar_bear_cafe.mp4",
        "subtitles": [
          {
            "chunksMetadata": {
              "count": 26,
              "endCueMs": 86430,
            },
            "id": "5f581055-dbf5-4f3b-bcee-a3215e405ffc",
            "streamIndex": 2,
            "type": "Embedded",
          },
          {
            "chunksMetadata": {
              "count": 15,
              "endCueMs": 86730,
            },
            "id": "94367f0c-92ef-4542-a80a-6fbaf8e1ad76",
            "name": "pbc_jp.ass",
            "type": "External",
          },
        ],
      },
      {
        "clips": [
          {
            "end": "00:08.76",
            "fields": {
              "meaning": "I am not a cat, said Frederick angrily",
              "transcription": "Ich bin keine Katze, sagte Frederick böse",
            },
            "id": "152077c6-336b-47d6-88e6-1ba4410db553",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:07.02",
            "tags": [
              "piggeldy_cat.mp4",
            ],
          },
          {
            "end": "00:20.0",
            "fields": {
              "meaning": "I didn't say that",
              "transcription": "Das hab' ich nicht gesagt",
            },
            "id": "864d822f-d482-438e-9620-684b2730a688",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:18.42",
            "tags": [
              "piggeldy_cat.mp4",
            ],
          },
        ],
        "duration": "00:58.24",
        "format": "mov,mp4,m4a,3gp,3g2,mj2",
        "id": "f5053503-8e5f-45c3-9afc-26f5e6eab8a8",
        "name": "piggeldy_cat.mp4",
      },
    ],
    "project": {
      "createdAt": "2020-02-01T16:10:37Z",
      "id": "ef3a2602-37a0-4158-89d6-47b75edb5bea",
      "name": "Project with subtitles",
      "noteType": "Transliteration",
      "timestamp": "2020-02-02T15:29:19Z",
    },
  },
}
`;
